### PR TITLE
AV-224167: Fix: AKO is crashing when hostrule, with LogAllHeader field in AnalyticsPolicy field, is applied to an ingress

### PR DESCRIPTION
**Pull request by akshayhavile**
…

### CODE REVIEW

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

File: tests/evhtests/l7_evh_crd_test.go [hunk 1]
```diff
@@ -722,23 +722,97 @@ func TestHostruleAnalyticsPolicyUpdateForEvh(t *testing.T) {
...
+
+	// update is not getting reflected on evh nodes immediately. Hence adding a sleep of 5 seconds.
+	time.Sleep(5 * time.Second)
+
+	// Check whether the AnalyticsPolicy values are properly updated
+	g.Expect(nodes[0].EvhNodes).To(gomega.HaveLen(1))
+	g.Expect(nodes[0].EvhNodes[0].AnalyticsPolicy).ShouldNot(gomega.BeNil())
+	g.Expect(*nodes[0].EvhNodes[0].AnalyticsPolicy.AllHeaders).To(gomega.BeTrue())
+
+	// Update host rule with AnalyticsPolicy - with LogAllHeader, Enabled
```
Comment (DixitAakash) [inline tests/evhtests/l7_evh_crd_test.go:760]: We can update this to :
Update host rule with AnalyticsPolicy - with LogAllHeader, FullClientLogs.Enabled

**Comment by akshayhavile** [comment]
build ako

**Review by arihantg — APPROVED** [review]
LGTM

**Comment by DixitAakash** [comment]
build ako

**Review by DixitAakash — APPROVED** [review]
LGTM

**Comment by akshayhavile** [comment]
build ako

**Comment by rachitshandilya** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Pull request merged**